src/jobs/processClinicalEvent.ts

```diff
@@ -38,7 +38,7 @@ const clinicalUpdateEvent = async (clinicalEvent: ClinicalUpdateEvent): Promise<
         logger.error(`Update Doc Error: ${e.stack}`);
       })
       .process(async file => {
-        const updatedFile = await recalculateFileState(file);
+        const updatedFile = await updateFileFromExternalSources(file);
```

Comment:
`updateFileFromExternalSources` is the new function that does the work previously done by `recalculateFileState` and also updates the file's `hasClinicalData` property. Every where in the code that was doing the recalculateFileState now gets to do the update from all external sources.